Target-MVD-Mathias/Views/TargetFormView.swift

```diff
@@ -0,0 +1,253 @@
+    
+    switch targetFormType {
+    case .creation:
+      delegate.saveTarget(area: area, title: title, topic: topic)
+    case .edition:
+      delegate.editTarget(area: area, title: title, topic: topic.label)
+    }
+  }
+  
+  // MARK: Functions
+  private func initView() {
+    loadNib()
+    setupTextFields()
+    setupSelectTopicButton()
+    setupLetterSpacing()
+    setupLabels()
+    disableSaveTargetButton()
+  }
+  
+  private func loadNib() {
+    Bundle.main.loadNibNamed("TargetFormView", owner: self, options: nil)
+    addSubview(contentView)
+    contentView.frame = self.bounds
+    contentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+  }
+  
+  private func setupTextFields() {
+    let textFields: [UITextField] = [areaLengthTextField, titleTextField]
+    UIHelper.stylizePlaceholdersFor(textFields, color: .black)
+    
+    areasPickerView.delegate = self
+    areasPickerView.dataSource = self
+    
+    areaLengthTextField.inputView = areasPickerView
+    
+    areasPickerView.selectRow(0, inComponent: 0, animated: false)
+    pickerView(areasPickerView, didSelectRow: 0, inComponent: 0)
+    
+    titleTextField.addLeftPadding()
+  }
+  
+  private func setupSelectTopicButton() {
+    selectTopicButton.addBorder(color: .black, weight: 1.0)
+  }
+  
+  private func setupLetterSpacing() {
+    cancelButton.titleLabel?.setSpacing(ofCharacter: 1.6)
+    saveTargetButton.titleLabel?.setSpacing(ofCharacter: 1.6)
+  }
+  
+  private func setupLabels() {
+    var areaLengthTitle: String
+    var topicTitle: String
+    
+    switch targetFormType {
+    case .creation:
+      areaLengthTitle = "SPECIFY AREA LENGTH"
+      topicTitle = "SELECT A TOPIC"
+    case .edition:
+      areaLengthTitle = "AREA LENGTH"
+      topicTitle = "TOPIC"
+    }
+    
+    areaLengthTitleLabel.text = areaLengthTitle
+    topicTitleLabel.text = topicTitle
+  }
+  
+  func resetFields() {
+    areasPickerView.selectRow(0, inComponent: 0, animated: false)
+    pickerView(areasPickerView, didSelectRow: 0, inComponent: 0)
+    
+    titleTextField.text = ""
+    title = ""
+    topic = Topic()
+  }
+  
+  func disableSaveTargetButton() {
+    saveTargetButton.isEnabled = false
+    saveTargetButton.layer.backgroundColor = UIColor.gray.cgColor
+  }
+  
+  func tryEnablingSaveTargetButton() {
+    let trimmedTitle = title.trimmingCharacters(in: .whitespaces)
+    
+    if !trimmedTitle.isEmpty && topic.id > 0 {
+      enableSaveTargetButton()
+    }
+  }
+  
+  func enableSaveTargetButton() {
+    saveTargetButton.isEnabled = true
+    saveTargetButton.layer.backgroundColor = UIColor.black.cgColor
+  }
+}
+
+extension TargetFormView: UITextFieldDelegate {
+  
+  func textFieldDidEndEditing(_ textField: UITextField) {
+    if textField == titleTextField {
+      title = titleTextField.text ?? ""
+    }
+  }
+  
+}
+
+extension TargetFormView: UIPickerViewDelegate {
+  
+  func numberOfComponents(in pickerView: UIPickerView) -> Int {
+    return 1
```

Comment:
I am not sure but probably this default value will be 1, test it and if works without please remove this method, if not just leave it.